codegen/templates/vec.rs.tera

```diff
@@ -389,8 +389,9 @@ impl {{ self_t }} {
 
     /// Creates a 4D vector from `self` and the given `w` value.
     #[inline]
-    pub fn extend(self, w: {{ scalar_t }}) -> {{ vec4_t }} {
-        {{ vec4_t }}::new(self.x, self.y, self.z, w)
+    pub const fn extend(self, w: {{ scalar_t }}) -> {{ vec4_t }} {
```

Comment:
Probably I should provide a SIMD implementation of Vec3A extend, which might be faster. It would mean this couldn't be const.